kmath-jafama/src/main/kotlin/space/kscience/kmath/jafama/KMathJafama.kt

```diff
@@ -0,0 +1,106 @@
+package space.kscience.kmath.jafama
+
+import space.kscience.kmath.operations.*
+import net.jafama.*
+
+/**
+ * A field for [Double] (using FastMath) without boxing. Does not produce appropriate field element.
+ */
+@Suppress("EXTENSION_SHADOWED_BY_MEMBER", "OVERRIDE_BY_INLINE", "NOTHING_TO_INLINE")
+public object JafamaDoubleField : ExtendedField<Double>, Norm<Double, Double>, ScaleOperations<Double> {
+    public override inline val zero: Double get() = 0.0
+    public override inline val one: Double get() = 1.0
+
+    public override fun number(value: Number): Double = value.toDouble()
+
+    public override fun binaryOperationFunction(operation: String): (left: Double, right: Double) -> Double =
+        when (operation) {
+            PowerOperations.POW_OPERATION -> ::power
+            else -> super<ExtendedField>.binaryOperationFunction(operation)
+        }
+
+    public override inline fun add(a: Double, b: Double): Double = a + b
+
+    public override inline fun multiply(a: Double, b: Double): Double = a * b
+    public override inline fun divide(a: Double, b: Double): Double = a / b
+
+    public override fun scale(a: Double, value: Double): Double = a * value
+
+    public override inline fun sin(arg: Double): Double = FastMath.sin(arg)
+    public override inline fun cos(arg: Double): Double = FastMath.cos(arg)
+    public override inline fun tan(arg: Double): Double = FastMath.tan(arg)
+    public override inline fun acos(arg: Double): Double = FastMath.acos(arg)
+    public override inline fun asin(arg: Double): Double = FastMath.asin(arg)
+    public override inline fun atan(arg: Double): Double = FastMath.atan(arg)
+
+    public override inline fun sinh(arg: Double): Double = FastMath.sinh(arg)
+    public override inline fun cosh(arg: Double): Double = FastMath.cosh(arg)
+    public override inline fun tanh(arg: Double): Double = FastMath.tanh(arg)
+    public override inline fun asinh(arg: Double): Double = FastMath.asinh(arg)
+    public override inline fun acosh(arg: Double): Double = FastMath.acosh(arg)
+    public override inline fun atanh(arg: Double): Double = FastMath.atanh(arg)
+
+    public override inline fun power(arg: Double, pow: Number): Double = FastMath.pow(arg, pow.toDouble())
+    public override inline fun exp(arg: Double): Double = FastMath.exp(arg)
+    public override inline fun ln(arg: Double): Double = FastMath.log(arg)
+
+    public override inline fun norm(arg: Double): Double = FastMath.abs(arg)
+
+    public override inline fun Double.unaryMinus(): Double = -this
```

Comment:
Does not make sense to override those.

Reply:
in basic arithmetic operations like these library functions are not even used should i just remove them instead?

Reply:
As long as you implement basic add/multiply it should work. They are present in DoubleField for performance considerations, But I am not sure that they matter.